src/main/java/se/liu/ida/hefquin/queryplan/executable/impl/ops/ExecOpGenericJoin.java

```diff
@@ -0,0 +1,43 @@
+package se.liu.ida.hefquin.queryplan.executable.impl.ops;
+
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import se.liu.ida.hefquin.data.jenaimpl.JenaBasedSolutionMapping;
+import se.liu.ida.hefquin.queryplan.executable.IntermediateResultBlock;
+import se.liu.ida.hefquin.queryplan.executable.IntermediateResultElementSink;
+import org.apache.jena.sparql.engine.join.JoinKey;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Collection;
+import java.util.List;
+
+public abstract class ExecOpGenericJoin implements BinaryExecutableOp {
+
+    protected Collection<Var> getVariables( final JenaBasedSolutionMapping JBSolMap){
+        final Binding b = JBSolMap.asJenaBinding();
```

Comment:
Please use the usual naming convention for variable and parameter names. They should begin with a lower case letter. Hence, instead of 'JBSolMap', it should be 'jbSolMap'.